pkg/webhook/workspace/config.go

```diff
@@ -31,32 +33,19 @@ func Configure(ctx context.Context) error {
 		return err
 	}
 
+	mutateWebhookCfg := buildMutateWebhookCfg()
+	validateWebhookCfg := buildValidatingWebhookCfg()
 	if !server.IsSetUp() {
-		log.Info("Webhooks server is not set up. Cleaning up webhook configurations")
-
-		if err := c.Delete(ctx, &v1beta1.MutatingWebhookConfiguration{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: mutateWebhookCfgName,
-			}}); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return err
-			}
-		}
-		if err = c.Delete(ctx, &v1beta1.ValidatingWebhookConfiguration{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: validateWebhookCfgName,
-			}}); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return err
-			}
-		}
+		_, mutatingWebhookErr := getMutatingWebhook(ctx, c, mutateWebhookCfg)
+		_, validatingWebhookErr := getValidatingWebhook(ctx, c, validateWebhookCfg)
 
-		log.Info("Existing workspace related webhook configurations are removed")
+		// No errors from either means that they are on the cluster
+		if mutatingWebhookErr == nil || validatingWebhookErr == nil {
+			return errors.New(`Webhooks have previously been set up and cannot be removed automatically. The admin must make sure that all workspaces are stopped, webhooks are removed, webhooks are reconfigured then restart the controller`)
```

Comment:
1. It makes sense to reformulate `webhooks are removed` something like `workspace-related webhooks configurations are removed`
2. `webhooks are reconfigured` could you clarify what do you mean here? Register `AdmissionWebhooksConfiguration` on the cluster? Or set `controller.webhooks.enabled` for controller to false? If so - isn't user expected to fall here only when webhooks are disabled?
```suggestion
			return errors.New(`Webhooks have previously been set up and cannot be removed automatically. Configure the controller to use webhooks again or make sure that all workspaces are stopped, webhooks configuration are removed and then restart the controller`)
```